packages/better-auth/src/oauth-2.1/types.ts

```diff
@@ -0,0 +1,319 @@
+import type { JWSAlgorithms } from "../plugins/jwt";
+
+/**
+ * Supported grant types of the token endpoint
+ */
+export type GrantType =
+	| "authorization_code"
+	// | "implicit" // NEVER SUPPORT - depreciated in oAuth2.1
+	// | "password" // NEVER SUPPORT - depreciated in oAuth2.1
+	| "client_credentials"
+	| "refresh_token";
+// | "urn:ietf:params:oauth:grant-type:device_code"  // specified in oAuth2.1 but not yet implemented
+// | "urn:ietf:params:oauth:grant-type:jwt-bearer"   // unspecified in oAuth2.1
+// | "urn:ietf:params:oauth:grant-type:saml2-bearer" // unspecified in oAuth2.1
+
+export type AuthMethod =
+	| "client_secret_basic" // Basic header
+	| "client_secret_post"; // POST
+// | "private_key_jwt" // must also add alg_values_supported for that endpoint
```

Comment:
Are these planned or for right now?

Reply:
Currently `client_secret_basic` and `client_secret_post` are supported as they are required in the OAuth 2.1 ref. `private_key_jwt` and `client_secret_jwt` are not planned in this release but it could in a future one.